packages/block-library/src/post-author/index.php

```diff
@@ -5,18 +5,136 @@
+	}
+
+	// Background color.
+	$has_named_background_color  = array_key_exists( 'backgroundColor', $attributes );
+	$has_custom_background_color = array_key_exists( 'customBackgroundColor', $attributes );
+
+	// If has background color.
+	if ( $has_custom_background_color || $has_named_background_color ) {
+		// Add has-background-color class.
+		$colors['css_classes'][] = 'has-background-color';
+	}
+
+	if ( $has_named_background_color ) {
+		// Add the background-color class.
+		$colors['css_classes'][] = sprintf( 'has-%s-background-color', $attributes['backgroundColor'] );
+	} elseif ( $has_custom_background_color ) {
+		// Add the custom background-color inline style.
+		$colors['inline_styles'] .= sprintf( 'background-color: %s;', $attributes['customBackgroundColor'] );
+	}
+
+	return $colors;
+}
+
+/**
+ * Build an array with CSS classes and inline styles defining the font sizes
+ * which will be applied to the navigation markup in the front-end.
+ *
+ * @param  array $attributes Navigation block attributes.
+ * @return array Font size CSS classes and inline styles.
+ */
+function post_author_build_css_font_sizes( $attributes ) {
+	// CSS classes.
+	$font_sizes = array(
+		'css_classes'   => array(),
+		'inline_styles' => '',
+	);
+
+	$has_named_font_size  = array_key_exists( 'fontSize', $attributes );
+	$has_custom_font_size = array_key_exists( 'customFontSize', $attributes );
+
+	if ( $has_named_font_size ) {
+		// Add the font size class.
+		$font_sizes['css_classes'][] = sprintf( 'has-%s-font-size', $attributes['fontSize'] );
+	} elseif ( $has_custom_font_size ) {
+		// Add the custom font size inline style.
+		$font_sizes['inline_styles'] = sprintf( 'font-size: %spx;', $attributes['customFontSize'] );
+	}
+
+	return $font_sizes;
+}
+
 /**
  * Renders the `core/post-author` block on the server.
  *
- * @return string Returns the filtered post author for the current post wrapped inside "h6" tags.
+ * @param  Object $block The block to render.
+ * @return string Returns the rendered author block.
  */
-function render_block_core_post_author() {
+function render_block_core_post_author( $block ) {
+
+	$attributes = $block->attributes;
+
+	if ( empty( $attributes ) ) {
+		return '';
+	}
+
 	$post = gutenberg_get_post_from_context();
+
 	if ( ! $post ) {
 		return '';
 	}
-	// translators: %s: The author.
-	return '<address>' . sprintf( __( 'By %s' ), get_the_author() ) . '</address>';
+
+	$avatar = ! empty( $attributes['avatarSize'] ) ? get_avatar(
+		$post->post_author,
+		$attributes['avatarSize']
+	) : null;
+
+	$byline     = ! empty( $attributes['byline'] ) ? $attributes['byline'] : false;
+	$colors     = post_author_build_css_colors( $attributes );
+	$font_sizes = post_author_build_css_font_sizes( $attributes );
+	$classes    = array_merge(
+		$colors['css_classes'],
+		$font_sizes['css_classes'],
+		array( 'wp-block-post-author' ),
+		isset( $attributes['className'] ) ? array( $attributes['className'] ) : array(),
```

Comment:
Could use the [null coalescing operator](https://www.php.net/manual/en/language.operators.comparison.php#language.operators.comparison.coalesce)

```suggestion
		 array( $attributes['className'] ) ?? array(),
```

Reply:
----------------------------------------------------------------------

 120 | ERROR | null coalescing operator (??) is not present in PHP

     |       | version 5.6 or earlier

🤦